manifests/init.pp

```diff
@@ -82,37 +82,39 @@
 #   Config Hash that is used to configure settings in logind.conf
 #
 class systemd (
-  Hash[String,Hash[String, Any]]                         $service_limits,
-  Boolean                                                $manage_resolved,
-  Enum['stopped','running']                              $resolved_ensure,
-  Optional[Variant[Array[String],String]]                $dns,
-  Optional[Variant[Array[String],String]]                $fallback_dns,
-  Optional[Variant[Array[String],String]]                $domains,
-  Optional[Variant[Boolean,Enum['resolve']]]             $llmnr,
-  Optional[Variant[Boolean,Enum['resolve']]]             $multicast_dns,
-  Optional[Variant[Boolean,Enum['allow-downgrade']]]     $dnssec,
-  Optional[Variant[Boolean,Enum['opportunistic', 'no']]] $dnsovertls,
-  Boolean                                                $cache,
-  Optional[Variant[Boolean,Enum['udp','tcp']]]           $dns_stub_listener,
-  Boolean                                                $use_stub_resolver,
-  Boolean                                                $manage_networkd,
-  Enum['stopped','running']                              $networkd_ensure,
-  Boolean                                                $manage_timesyncd,
-  Enum['stopped','running']                              $timesyncd_ensure,
-  Optional[Variant[Array,String]]                        $ntp_server,
-  Optional[Variant[Array,String]]                        $fallback_ntp_server,
-  Boolean                                                $manage_accounting,
-  Hash[String,String]                                    $accounting,
-  Boolean                                                $purge_dropin_dirs,
-  Boolean                                                $manage_journald,
-  Systemd::JournaldSettings                              $journald_settings,
-  Boolean                                                $manage_logind,
-  Systemd::LogindSettings                                $logind_settings,
+  Optional[Hash[String,Hash[String, Any]]]               $service_limits      = undef,
```

Comment:
The default for hashes and arrays should always be an emtpy hash / array. That's the case in the master branch. Why whould you change that?

Reply:
Hello,

I doesn't like empty value, It more memory and CPU usage, but I can change if you want.

Reply:
Generally speaking, `undef` should only be the default for strings, as other types have more sensible defaults. Using `undef` as default for enumerable types results in unnecessarily complex code such as used below (`if` instead of a simple loop on an empty set).

Reply:
Btw, this also means enumerable types should not be marked as `Optional`.